Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -514,16 +514,23 @@ public static void Update()
             // update all server objects
             foreach (KeyValuePair<uint, NetworkIdentity> kvp in NetworkIdentity.spawned)
             {
-                NetworkIdentity identity = kvp.Value;
-                if (identity != null)
+                // try to update.
+                // spawned might have null entries if the user made a mistake.
+                //
+                // IMPORTANT: do not check kvp.Value != null. this is way too
+                //            costly due to Unity's custom null check
+                //            instead, catch the exception if it happens.
+                //            (see 4k benchmark).
+                try
                 {
-                    identity.ServerUpdate();
+                    kvp.Value.ServerUpdate();
                 }
-                else
+                // spawned list should have no null entries because we
+                // always call Remove in OnObjectDestroy everywhere.
+                // but if it does, we should let the user know how it happens
+                catch (NullReferenceException)
```

Comment:
```suggestion
                catch (MissingReferenceException )
```
unity throws MissingReferenceException instead of NullReferenceException. The only time an object should be `"null"` here is if it was destroyed but has not yet been removed from spawned